plugins/cython/compile_cython.py

```diff
@@ -1,33 +1,40 @@
 import os
 import shutil
-import subprocess
 
 from pants.backend.python.targets.python_library import PythonLibrary
-from pants.backend.python.tasks.python_execution_task_base import PythonExecutionTaskBase
+from pants.backend.python.tasks import pex_build_util
+from pants.backend.python.tasks.resolve_requirements import ResolveRequirements
 from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError
 from pants.base.payload import Payload
+from pants.base.payload_field import PrimitiveField
 from pants.build_graph.resources import Resources
 from pants.task.simple_codegen_task import SimpleCodegenTask
+from pants.util.contextutil import temporary_dir
 from pants.util.dirutil import safe_mkdir_for
+from pex.interpreter import PythonInterpreter
+from pex.pex import PEX
+from pex.pex_builder import PEXBuilder
 from pex.pex_info import PexInfo
 
 
 class CompileCython(PythonLibrary):
-    def __init__(self, inputs=None, payload=None, output=None, address=None, **kwargs):
-        inputs = inputs.legacy_globs_class.create_fileset_with_spec(address.spec_path, *inputs.filespecs['globs'])
+    def __init__(self, payload=None, address=None, output=None, **kwargs):
         payload = payload or Payload()
-        payload.add_fields({
-            'inputs': self.create_sources_field(inputs, sources_rel_path=address.spec_path, key_arg='sources')
-        })
-        self.output = output
+        payload.add_field('output', PrimitiveField(output))
         super(CompileCython, self).__init__(payload=payload, address=address, **kwargs)
 
+    @property
+    def output(self):
+        return self.payload.output
+
 
-class CompileCythonCreate(SimpleCodegenTask, PythonExecutionTaskBase):
```

Comment:
There were two root problems working together against your plugin.
1. installation in the 'compile' goal, which, in the new python pipeline, is too late for the `GatherSources` (in 'pyprep' hidden goal) task that `PytestRun` via `PytestPrep` uses to form test chroots.
2. Mixing in `PythonExecutionTaskBase` whose [product requirements](https://github.com/pantsbuild/pants/blob/master/src/python/pants/backend/python/tasks/python_execution_task_base.py#L34-L39) for it to run after the 'pyprep' hidden goal and thus leads to an impossible to install task.